app/src/main/java/com/example/poste/activities/DashboardActivity.java

```diff
@@ -195,15 +211,43 @@ public void onClick(DialogInterface dialog, int which) {
                 }
 
                 // Handle post creation logic
-                try {
-                    // Create the post in the API
-                    int newPostID = API.addPost(itemName, itemLink, currentUser.getId());
-
-                    // Update the post's folder in the API
-                    API.addPostToFolder(newPostID, selectedFolderId);
-                } catch (APIException e) {
-                    throw new RuntimeException(e);
-                }
+                Call<ResponseBody> call = apiService.createPost("Token " + currentUser.getToken(), new PostRequest(itemName, "", itemLink, selectedFolderId));
+
+                call.enqueue(new Callback<ResponseBody>() {
+                    @Override
+                    public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                        if(response.isSuccessful()) {
+                            try {
+                                JSONObject res = new JSONObject(response.body().toString());
+
+                                if (res.getBoolean("success")) {
+                                    // finds the local version of the folder.
+                                    com.example.poste.models.Folder folder = null;
+                                    for (com.example.poste.models.Folder find: currentUser.getFolders()) {
+                                        if (Integer.parseInt(find.getId()) == res.getInt("folder")) {
+                                            folder = find;
+                                        }
+                                    }
+
+                                    // create a local post matching the one created on the API
+                                    currentUser.updateFoldersAndPosts(DashboardActivity.this);
+                                    Toast.makeText(DashboardActivity.this, "Post creation successful.", Toast.LENGTH_LONG).show();
+                                } else {
+                                    Toast.makeText(DashboardActivity.this, "Post creation failed.", Toast.LENGTH_LONG).show();
+                                }
+                            } catch (JSONException e) {
+                                throw new RuntimeException(e);
```

Comment:
Careful with this -- if the response isn't handled appropriately, this just crashes the app (which it just did to me in creating a folder)

Might be beneficial to try to handle it gracefully -- logging, printing a Toast, or something, and trying to restore to a stable state

Reply:
changed to a printout. Removed 2 since I don't need to convert the response body to a JSON object any more.

Reply:
This will work. I'd also suggest a Logged message, so we can comb through the logger to see what response we got. This will help us implement more graceful error handling and in debugging potential problems down the road.

I created some logging utils; check around other PR's if you want to see some examples, or feel free to add your own. Logging is an important part of the testing / debug process, and if something is failing here, it's supremely helpful to know what we got back and why it didn't work (sometimes its as simple as not properly parsing out the returned response, when it actually succeeded)